src/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -70,11 +71,27 @@ internal static bool IsDirectoryTooLong(string fullPath)
         }
 
         /// <summary>
-        /// Adds the extended path prefix (\\?\) if not already present.
+        /// Adds the extended path prefix (\\?\) if not already present, IF the path is not relative,
+        /// AND the path is more than 259 characters. (> MAX_PATH + null)
+        /// </summary>
+        internal static string AddExtendedPathPrefixForLongPaths(string path)
```

Comment:
Nit: Personally, I'd prefer it if the name of the method made it clear that it might not add anything, e.g. AddLongPathExtendedPrefixIfNecessary.